ansible/roles/post-recheck-comment/tasks/main.yaml

```diff
@@ -0,0 +1,64 @@
+---
+
+- name: Store Verified-1 regex pattern
+  set_fact:
+    failure_pattern: '^Patch Set {{ patchset }}: Verified-1'
+
+- name: Wait for Verified-1 comment from {{ ci_user }}
+  register: comments
+  raw: gerrit query change:{{ change_id }} --comments --format JSON
+  retries: 6
+  delay: 10
+  until: >
+    comments.rc == 0 and
+    (comments.stdout_lines[0] | from_json).comments
+    | selectattr('reviewer.username', 'equalto', ci_user)
+    | map(attribute='message')
+    | list
+    | last
+    | regex_search(failure_pattern)
+  tags:
+    # This would fire [ANSIBLE0012] Commands should not change things if nothing needs doing,
+    # but it's ok as this command is side-effect free.
+    - skip_ansible_lint
+
+- name: Count failures
+  set_fact:
+    n_failures: >
+      {{
+      (comments.stdout_lines[0] | from_json).comments
+      | selectattr('reviewer.username', 'equalto', ci_user)
+      | map(attribute='message')
+      | map('regex_search', failure_pattern)
+      | select
+      | list
+      | count
+      }}
+
+- name: Show failure count
+  debug:
+    var: n_failures
+
+- name: Prepare the recheck message
+  when: n_failures | int < max_rechecks | int + 1
+  set_fact:
+    comment_message: >
+      We've detected that Windows CI failed due to flakiness, so we're going to recheck windows
```

Comment:
We were thinking `n_failures` logic. Right now it counts how many `Verified: -1` comments were posted, but it takes into account each CI failure - flakiness and compilation errors for example.

Maybe we should calculate `n_failures` as an amount of comments containing `recheck windows`. What do you think?

Reply:
> flakiness and _compilation errors_ for example.

I'm not sure I understand, if the compilation error had occurred, then the author must have submitted a new patchset to fix it. And we're looking at the `-1` comments only for a specific patchset, so the older failures doesn't count at all.

Reply:
Ok, makes sense. We misunderstood the part about patchsets.